brontide/read_buffer_pool.go

```diff
@@ -0,0 +1,85 @@
+package brontide
+
+import (
+	"time"
+
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/queue"
+)
+
+const (
+	// DefaultGCInterval is the default interval that the WriteBufferPool
+	// will perform a sweep to see which expired buffers can be released to
+	// the runtime.
+	DefaultGCInterval = 15 * time.Second
+
+	// DefaultExpiryInterval is the default, minimum interval that must
+	// elapse before a WriteBuffer will be released. The maximum time before
+	// the buffer can be released is equal to the expiry interval plus the
+	// gc interval.
+	DefaultExpiryInterval = 30 * time.Second
+
+	// readBufferSize represents the size of the maximum message that can be
+	// read off  the wire by brontide. The buffer is used to hold the
+	// ciphertext while the  brontide state machine decrypts the message.
+	readBufferSize = lnwire.MaxMessagePayload + macSize
+)
+
+// ReadBuffer is a static byte array sized to the maximum-allowed message size,
+// plus 16 bytes for the MAC.
+type ReadBuffer [readBufferSize]byte
+
+// Recycle zeroes the ReadBuffer, making it fresh for another use.
```

Comment:
Possibly make this a function instead so the block can be re-used rather than copied between contexts.